iceoryx_hoofs/include/iceoryx_hoofs/internal/memory/pointer_repository.inl

```diff
@@ -46,7 +46,7 @@ template <typename id_t, typename ptr_t, uint64_t CAPACITY>
 inline cxx::optional<id_t> PointerRepository<id_t, ptr_t, CAPACITY>::registerPtr(const ptr_t ptr,
                                                                                  const uint64_t size) noexcept
 {
-    for (id_t id = 1U; id <= MAX_ID; ++id)
+    for (id_t id{1U}; id <= MAX_ID; ++id)
```

Comment:
Bad rule. Not wrong, but requiring this style for primitive types is a bad idea, even more so in loops.

Resolve.